apps/libexec/merlinkey.py.in

```diff
@@ -17,27 +17,44 @@ def cmd_generate(args):
 	"""
 
 	path = merlin_conf_dir
+	write_to_file = False
 	for arg in args:
 		if arg.startswith('--path'):
 			path = arg.split('=', 1)[1]
+		if arg.startswith('--write'):
+			write_to_file = True
 
-	if os.path.isfile(path + '/key.pub'):
-		print 'File already exists. Key will not be generated.'
-		sys.exit(1)
+	has_pub_key = os.path.isfile(path + '/key.pub')
+	has_priv_key = os.path.isfile(path + '/key.priv')
+
+	if has_pub_key and has_priv_key:
+		print 'File already exists. Keys will not be generated.'
+		sys.exit(0)
 
-	if os.path.isfile(path + '/key.priv'):
-		print 'File already exists. Key will not be generated.'
+	if has_pub_key or has_priv_key:
+		print 'One of the encryption keys already exists. Remove and try again'
```

Comment:
From a logging perspective I think it would be nice if the error specified which of the keys exist, rather than "one of them exist".

Reply:
I agree, but I will not do this.